modules/uhsg_domain/src/Plugin/Block/LinkToInstructionsForStudents.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Drupal\uhsg_domain\Plugin\Block;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * Provides a 'link_to_instructions_for_students' block.
+ *
+ * @Block(
+ *  id = "link_to_instructions_for_students",
+ *  admin_label = @Translation("Link to Instructions for students"),
+ * )
+ */
+class LinkToInstructionsForStudents extends BlockBase {
+
+  use StringTranslationTrait;
+
+  /**
+   * @inheritdoc
+   */
+  public function access(AccountInterface $account, $return_as_object = FALSE) {
+    return \Drupal::service('uhsg_domain.domain')->isTeachingDomain()
+      ? AccessResult::allowed()
+      : AccessResult::forbidden();
```

Comment:
I would do here a `!\Drupal::service('uhsg_domain.domain')->isStudentDomain()`, because we probably would want to show this link "because we are not at the student domain" instead "because we are at teacher domain". If third domain would be involved this condition would likely make more sense.

Reply:
Agreed, modifying.